python/ray/_common/usage/usage_lib.py

```diff
@@ -728,18 +728,23 @@ def cloud_metadata_request(url: str, headers: Optional[Dict[str, str]]) -> bool:
         return False
 
     # Make internal metadata requests to all 3 clouds
-    if cloud_metadata_request(
-        "http://metadata.google.internal/computeMetadata/v1",
-        {"Metadata-Flavor": "Google"},
-    ):
+    # Check Azure FIRST before AWS since they share the same IP address (169.254.169.254)
+    # and AWS's generic endpoint might return non-404 errors from Azure IMDS
+    AZURE_METADATA_URL = (
+        "http://169.254.169.254/metadata/instance?api-version=2021-12-13"
+    )
```

Comment:
Actually is this the preferred way to check if we are inside azure or not? @jackfrancis @marosset